skyportal/handlers/api/source.py

```diff
@@ -87,7 +87,8 @@ def head(self, obj_id=None):
         if num_s > 0:
             return self.success()
         else:
-            return self.error()
+            self.set_status(404)
```

Comment:
Candidate handler probably also needs this

Reply:
Yes, please add the same for candidates.HEAD.

Reply:
Done.